### DISCUSSION TITLE
Upgrade from 3.5 to higher version

### DISCUSSION THREAD

**Discussion by santoshc1306** [ravendb/ravendb#19233] [Q&A]
Hi, Is it possible to upgrade from raven version 3.5 to version 6 directly ? 
 Can we setup replication between 3.5 and 6 ? 
What are the options available to upgrade without application downtime.

**Comment by ayende** [comment]
You can setup replication from 3.5 to 6.0 - yes.
The process of migration from 3.5 to 6.0 is identical to moving from 3.5 to 4.0.
See:

https://ravendb.net/docs/article-page/4.0/csharp/migration/client-api/introduction

**Reply by grisha-kotler** [reply]
To enable replication to a secured RavenDB 4.x/5.x/6.x server, you need to modify the RavenDB 3.x configuration file and add the following options:
- Raven/Replication/CertificatePath
- Raven/Replication/CertificatePassword (optional)

The certificate must be generated from the 4.x/5.x/6.x server and include the necessary permissions for the required databases. (Use the Operator role if you need to migrate indexes as well.)

Please note that this requires the latest version of RavenDB 3.5, as this functionality was added in recent updates.

**Comment by santoshc1306** [comment]
The server build we are on is currently 35180 , so that might require an
upgrade before we set up the replication to a higher version.
Can you provide which version of 3.5 has the updates

On Mon, Sep 16, 2024 at 9:31 AM Grisha Kotler ***@***.***>
wrote:

> To enable replication to a secured RavenDB 4.x/5.x/6.x server, you need to
> modify the configuration file and add the following options:
>
>    - Raven/Replication/CertificatePath
>    - Raven/Replication/CertificatePassword (optional)
>
> The certificate must be generated from the 4.x/5.x/6.x server and include
> the necessary permissions for the required databases. (Use the Operator
> role if you need to migrate indexes as well.)
>
> Please note that this requires the latest version of RavenDB 3.5, as this
> functionality was added in recent updates.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ravendb/ravendb/discussions/19233#discussioncomment-10656830>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2SYBWWLCXFJBNXMTYMZPPTZW2JNJAVCNFSM6AAAAABNYT57HCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANRVGY4DGMA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Regards,
Santosh C.

**Reply by grisha-kotler** [reply]
`3.5.10-patch-35294`, but if IIRC, there were additional changes made afterward to ensure this works properly.
